amethyst_assets/src/progress.rs

```diff
@@ -0,0 +1,143 @@
+use std::sync::Arc;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// Completion status, returned by `ProgressCounter::complete`.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Completion {
+    /// Loading is complete
+    Complete,
+    /// Some asset loads have failed
+    Failed,
+    /// Still loading assets
+    Loading,
+}
+
+/// The `Progress` trait, allowing to track which assets are
+/// imported already.
+pub trait Progress {
+    /// The tracker this progress can create.
+    type Tracker: Tracker;
+
+    /// Add `num` assets to the progress.
+    /// This should be done whenever a new asset is
+    /// put in the queue.
+    fn add_assets(&mut self, num: usize);
+
+    /// Creates a `Tracker`.
+    fn create_tracker(self) -> Self::Tracker;
+}
+
+impl Progress for () {
```

Comment:
So basically now you can also pass `()` if you don't care and just want the assets to pop in once they're there.

Reply:
Error handling needs improvements, but I think one can see that this design at least allows them, thus we should do that in a future PR.